.github/workflows/ci_catalyst.yml

```diff
@@ -0,0 +1,122 @@
+name: Catalyst CI
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "mlir/**"
+  pull_request:
+    paths:
+      - "mlir/**"
+      - ".github/workflows/ci_catalyst.yml"
+  merge_group:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  cpp-test-catalyst:
+    name: 🇨‌ Test MLIR with LLVM@${{ matrix.llvm-version }}
+    runs-on: macos-14
+    strategy:
+      matrix:
+        llvm-version: [19]
```

Comment:
Would be interesting to see this work on `macos-13` (x86 runners) as well as `llvm-version: 20`.
The current build pipeline seems to pass, which is great already!

Unrelated to this PR in particular, but what could be a nice byproduct in a separate PR is that this already represents a functional macOS-14 CI pipeline for MLIR, which would be part of #925.